README.md

```diff
@@ -0,0 +1,128 @@
+# GKE Service Accounts Initializer
+
+> :warning: **This is pre-release software. Do not use it for production
+> purposes.**
+
+This add-on inject [Google Cloud Service Accounts][sa] credentials, which are
+previously imported to the cluster as Secrets, to [Pods] running on Container
+Engine.
+
+[sa]: https://cloud.google.com/compute/docs/access/service-accounts
+[Pods]: https://kubernetes.io/docs/concepts/workloads/pods/pod/
+[GKE]: https://cloud.google.com/container-engine/
+
+Install this initializer to your cluster and add the following annotation to
+metadata.annotations field of your Pod template or specification:
+
+```yaml
+annotations:
```

Comment:
Is the security model like this: authz/authn will ensure only qualified user can create the pod, so the initializer should trust the annotation and append the secret as asked?

Reply:
I think this fits the model you described, however I'm not sure if it's the best practice. (Read this like I don't know much about how k8s RBAC works.)

If I'm a user who has no access to read/use a Secret, can I still create a Pod that has k8s ServiceAccount that can read/use the secret? (We might be unintentionally elevating privileges by mounting the secret with Initializer account which runs like root in the cluster).

It might be good to cc some people to double-check this. Can you suggest anyone?

<small>P.S. Also I realized I need to provide clusterrolebinding for this initializer, it doesn't work out-of-the box currently on kubeadm clusters. That'll be another PR.</small>

Reply:
> If I'm a user who has no access to read/use a Secret, can I still create a Pod that has k8s ServiceAccount that can read/use the secret? (We might be unintentionally elevating privileges by mounting the secret with Initializer account which runs like root in the cluster).

Discussed with @tallclair, it seems the serviceaccount admission plugin [restricts](https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/admission/serviceaccount/admission.go#L189) what secrets can be referenced by a pod. After your intiailizer injects the secret, the plugin will check if the pod's service account allows referencing that secret.

Reply:
Got it, I think we're good then. Anyone who can run a pod that has access to the secret means they can export the secret anyway.

Reply:
To clarify, the SA admission plugin has that capability, but it's not currently enabled (nor is there a flag exposed to enable it). However, pods can only reference secrets in the same namespace.